hackathon_site/dashboard/frontend/src/components/general/ProductOverview/ProductOverview.test.js

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import SideSheetRight from "../SideSheetRight/SideSheetRight";
+import { ProductOverview, EnhancedCartForm } from "./ProductOverview";
+import {
+    render,
+    fireEvent,
+    waitForElementToBeRemoved,
+    waitFor,
+    waitForElement,
+} from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+describe("Product Overview", () => {
+    const testDetail = {
+        name: "Arduino",
+        type: "red",
+        total: 30,
+        available: 29,
+        img: "https://www.filterforge.com/more/help/images/size200.jpg",
+        category: ["MCU", "FPGA"],
+        manufacturer: "Canakit",
+        model_num: "Model 3B+",
+        datasheet: "link",
+        notes: ["- For micropython ask for image", "- randomnerdtutorials.com"],
+        constraints: ["- Max 1 of this item", "- Max 3 microcontroller labelled red"],
+        quantity: 3,
+    };
+
+    test("render", () => {
+        const addCartMock = jest.fn();
+
+        const { asFragment } = render(
+            <SideSheetRight title="Product Overview" isVisible={true}>
+                <ProductOverview detail={testDetail} addCartFunction={addCartMock} />
+            </SideSheetRight>
+        );
+
+        expect(asFragment()).toMatchSnapshot();
+    });
+
+    test("add to cart button calls the correct function", () => {
+        const addCartMock = jest.fn();
+
+        const { getByText, getByRole } = render(
+            <SideSheetRight title="Product Overview" isVisible={true}>
+                <ProductOverview detail={testDetail} addToCart={addCartMock} />
+            </SideSheetRight>
+        );
+
+        const button = getByText("ADD TO CART");
+
+        fireEvent.click(button);
+        expect(addCartMock).toHaveBeenCalledTimes(2);
+    });
+});
+
+describe("<EnhancedCartForm />", () => {
```

Comment:
The tests this will wind up needing:
- Doesn't call handleSubmit when an option isn't selected, and displays the error message
- When a value is selected, it submits and calls `handleSubmit` with the correct values (you're close on this one).

Reply:
@lisa-sa-li mentioned that the default option is 1

Reply:
Sounds good, as long as there's no way to actually clear the value then you can skip that test (but I would still include it in a yup schema, just for completeness).